$MirOS: ports/comms/linphone/patches/patch-console_linphonec_c,v 1.1.7.1 2005/03/18 15:43:50 tg Exp $
--- console/linphonec.c.orig	Fri Nov 28 09:41:57 2003
+++ console/linphonec.c	Wed Oct  6 00:58:07 2004
@@ -123,6 +123,7 @@ linphonec_main_loop (LinphoneCore * opm)
 {
 	char input[256], ch;
 	char temp[256];
+	char *et;
 	int i;
 	int err;
 
@@ -132,7 +133,13 @@ linphonec_main_loop (LinphoneCore * opm)
 	{
 		fprintf (stdout, "\nCommand ? ");
 		fflush (stdout);
-		fgets (input, 255, stdin);
+	eagainloop:	/* XXX yes this is weird, but hey, it works */
+		et = fgets (input, 255, stdin);
+		if ((et == NULL) && (errno == EAGAIN)) {
+			usleep(100000);
+			goto eagainloop;
+		}
+
 		ch = (char) tolower (input[0]);
 
 		switch (ch)
@@ -203,6 +210,9 @@ linphonec_main_loop (LinphoneCore * opm)
 				send_dtmf (opm->audiostream, ch);
 			break;
 
+		case '\n':
+		case '\0':
+			break;
 		default:
 			fprintf (stdout,
 				 "\nHelp:\nr\tregister\nc <url>\tmake call\nt\tterminate call\na\tanswer call\nq\tquit\n1-9#*\tsend dtmf\n");
